### PR TITLE
Add a dotColor property

### PR DESCRIPTION
**Pull request by TeaSeaLancs**
Adds a `dotColor` parameter which can be used to restyle the dots in situations where you need a light colour background.

I’ve also made a similar change for material-auto-rotate-carousel to pass down that property, i’ll submit once that PR if/when this one is accepted :)

### CODE REVIEW

**Comment by TeaSeaLancs** [comment]
(Apologies for the weirdness on this PR, my rebasing attempts didn't go according to plan! Everything is fine now)

File: src/Dots.js [hunk 1]
```diff
@@ -74,7 +83,7 @@ export default class Dots extends Component {
                 circle
                 zDepth={0}
                 style={{
-                  ...styles.dot,
+                  ...dotStyle,
```
Comment (leMaik) [inline src/Dots.js]: Just using
```
...styles.dot,
color: dotColor
```
would make all the weird state stuff and the added prepare function unneccessary.

File: src/Dots.js [hunk 2]
```diff
@@ -83,7 +92,7 @@ export default class Dots extends Component {
           <Paper
             zDepth={0}
             style={{
-              ...styles.dot,
+              ...dotStyle,
```
Comment (leMaik) [inline src/Dots.js]: see above

File: src/Dots.spec.js [hunk 3]
```diff
@@ -48,3 +48,14 @@ test('onDotClick handler', () => {
...
+
+test('custom dot color', () => {
+  const tree = renderer.create(
+    <ThemedDots
+      count={4}
+      index={0}
+      dotColor='red'
+    />
+  )
+  expect(tree).toMatchSnapshot()
```
Comment (leMaik) [inline src/Dots.spec.js:60]: It would be better if this test would actually check if the dots have the specified color.

**Review by leMaik — CHANGES_REQUESTED** [review]
This approach is far too complicated. :open_mouth: See my first comment on how to make this much simpler.

**Comment by leMaik** [comment]
@TeaSeaLancs Great enhancement :+1:

**Comment by TeaSeaLancs** [comment]
In my defence my intentions were good, I just didn't follow through as purely as I could have otherwise done. The reason why I did it the way I did it was to prevent object & update thrashing by constantly passing in new style objects, but we're already not doing that anyway so what was the point? I'll change it.

File: src/Dots.spec.js (hunk 3 above)
Comment (TeaSeaLancs) [reply src/Dots.spec.js:60]: Technically the snapshot testing would? If you'd like an explicit test, is there a way to do it without having to set up enzyme in the test environment?

**Comment by leMaik** [comment]
> The reason why I did it the way I did it was to prevent object & update thrashing by constantly passing in new style objects, but we're already not doing that anyway so what was the point? I'll change it.

Let's not bother about micro optimization in a tiny component that will certainly only used once per page. :wink:

File: src/Dots.spec.js (hunk 3 above)
Comment (leMaik) [reply src/Dots.spec.js:60]: You're right, the snapshot test checks this already. :+1:

**Review by leMaik — APPROVED** [review]

**Comment by TeaSeaLancs** [comment]
Apologies for writing on a closed PR, but i'm wondering, is it possible to get this enhancement into an official release prior to version 2?

**Comment by leMaik** [comment]
> Apologies for writing on a closed PR

Actually, _thank you_ for the reminder, I totally forgot this one. :sweat_smile:
:package: v1.2.0 is out now

**Comment by leMaik** [comment]
@TeaSeaLancs 
> Apologies for writing on a closed PR

Actually, _thank you_ for the reminder, I totally forgot this one. :sweat_smile:
:package: v1.2.0 is out now

**Pull request merged**